pkg/traceql/engine.go

```diff
@@ -238,3 +355,48 @@ func (s Static) asAnyValue() *common_v1.AnyValue {
 		},
 	}
 }
+
+// Regex to extract matchers from a query string
+// This regular expression matches a string that contains three groups separated by operators.
+// The first group is a string of alphabetical characters, dots, and underscores.
+// The second group is a comparison operator, which can be one of several possibilities, including =, >, <, and !=.
+// The third group is one of several possible values: a string enclosed in double quotes,
+// a number with an optional time unit (such as "ns", "ms", "s", "m", or "h"),
+// a plain number, or the boolean values "true" or "false".
+// Example: "http.status_code = 200" from the query "{ .http.status_code = 200 && .http.method = }"
+var matchersRegexp = regexp.MustCompile(`[a-zA-Z._]+\s*[=|<=|>=|=~|!=|>|<|!~]\s*(?:"[a-zA-Z._-]+"|[0-9smh]+|true|false)`)
+
+// TODO: Merge into a single regular expression
+
+// Regex to extract selectors from a query string
+// This regular expression matches a string that contains a single selector and no OR `||` conditions.
```

Comment:
Let's change `selector` to `spanset filter` which matches the traceql definitions, and add a few examples of queries that match or don't match.